frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -234,94 +235,99 @@ function toRequestPayload(
     return request;
 }
 
-async function handleSubmit() {
-    if (forestClientData.value.length > 0) {
-        const successList: string[] = [];
-        const warningList: string[] = [];
-        const errorList: string[] = [];
-        for (const item of forestClientData.value) {
-            const data = toRequestPayload(formData.value, item);
-
-            await userRoleAssignmentApi
-                .createUserRoleAssignment(data as FamUserRoleAssignmentCreate)
-                .then(() => {
-                    successList.push(item.forest_client_number);
-                })
-                .catch((error) => {
-                    if (error.response?.status === 409) {
-                        warningList.push(item.forest_client_number);
-                    } else {
-                        errorList.push(item.forest_client_number);
-                    }
-                });
-        }
+const composeAndPushNotificationMessages = (
+    successIdList: string[],
+    warningIdList: string[],
+    errorIdList: string[]
+) => {
+    const username = formData.value.userId.toUpperCase();
+    if (successIdList.length > 0) {
+        pushNotification(
+            Severity.success,
+            `${username} ${
+                successIdList[0] === ''
+                    ? `was successfully added with the role ${
+                          getSelectedRole()?.role_name
+                      }`
+                    : `was successfully added with Client IDs: ${successIdList.join(
+                          ', '
+                      )}`
+            }`
+        );
+    }
+    if (warningIdList.length > 0) {
+        pushNotification(
+            Severity.warning,
+            `${username} ${
+                warningIdList[0] === ''
+                    ? `already exists with the role ${
+                          getSelectedRole()?.role_name
+                      }`
+                    : `already exists with Client IDs: ${warningIdList.join(
+                          ', '
+                      )}`
+            }`
+        );
+    }
+    if (errorIdList.length > 0) {
+        pushNotification(
+            Severity.error,
+            `An error has occured. ${username} ${
+                errorIdList[0] === ''
+                    ? `could not be added with the role ${
+                          getSelectedRole()?.role_name
+                      }`
+                    : `was not added with Client IDs: ${errorIdList.join(', ')}`
+            }`
+        );
+    }
+    return '';
+};
+
+const handleSubmit = async () => {
+    const submittedSuccessList: string[] = [];
+    const submittedWarningList: string[] = [];
+    const submittedErrorList: string[] = [];
```

Comment:
Could we use `successForestClientIdList`, `warningForestClientIdList`, `errorForestClientIdList`? No need for `submitted`.

Reply:
Thanks Ian! Changed as suggested.